qiita_pet/uimodules/preprocessed_data_tab.py

```diff
@@ -57,10 +56,14 @@ def render(self, study_id, preprocessed_data):
         # New Type is for users to add a new user-defined investigation type
         user_defined_terms = ontology.user_defined_terms + ['New Type']
 
-        if PrepTemplate.exists(preprocessed_data.prep_template):
-            prep_template_id = preprocessed_data.prep_template
-            prep_template = PrepTemplate(prep_template_id)
-            raw_data_id = prep_template.raw_data
+        # ppd can only have 1 prep template
+        prep_template = preprocessed_data.prep_templates[0]
+        # this block might seem wrong but is here due to a possible
```

Comment:
This is not gonna work as it is right now. `prep_template` is already a PrepTemplate so if we have that pathological case, it will break in the call above `preprocessed_data.prep_templates[0]`, as it could not instantiate any prep template, so you'll receive an IndexError. If this pathological case still exists (which it shouldn't) the correct solution is:

```python
prep_templates = preprocessed_data.prep_templates
if len(prep_templates == 1):
    prep_template_id = prep_template.id
    raw_data_id = prep_template.artifact.id
    inv_type = prep_template.investigation_type or "None selected"
else:
    prep_template_id = None
```